src/aosm/azext_aosm/_params.py

```diff
@@ -0,0 +1,153 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from argcomplete.completers import FilesCompleter
+from azure.cli.core import AzCommandsLoader
+
+from .util.constants import CNF, VNF, BICEP_PUBLISH, ARTIFACT_UPLOAD, IMAGE_UPLOAD
+
+
+def load_arguments(self: AzCommandsLoader, _):
+    from azure.cli.core.commands.parameters import (
+        file_type,
+        get_enum_type,
+        get_three_state_flag,
+    )
+
+    definition_type = get_enum_type([VNF, CNF])
+    nf_skip_steps = get_enum_type([BICEP_PUBLISH, ARTIFACT_UPLOAD, IMAGE_UPLOAD])
+    ns_skip_steps = get_enum_type([BICEP_PUBLISH, ARTIFACT_UPLOAD])
+
+    # Set the argument context so these options are only available when this specific command
+    # is called.
+
+    with self.argument_context("aosm nfd") as c:
+        c.argument(
+            "definition_type", arg_type=definition_type, help="Type of AOSM definition."
+        )
```

Comment:
is this param required for users when using all `aosm nfd` commands? If so, please add `required=True` for this param.
After adding this restriction, combining with `arg_type=definition_type`, users can only input specific values, and it is not necessary to validate the users' input like https://github.com/jddarby/azure-cli-extensions/blob/c172f412bf91516a726f1ec74ff8ee3ca5a9b534/src/aosm/azext_aosm/custom.py#L252.

Reply:
Fixed